### PR TITLE
Allow apostrophe in venv name

### LINKED ISSUE #8947: Uv fails to activate environment in folder with apostrophe
I tried these commands in a folder with an apostrophe and a space.
```shell
uv venv
source .venv/bin/activate
```
Version: 0.4.25
Installation Method: Dnf package 
OS: Fedora Linux 41 KDE Spin

### PR DESCRIPTION
**Pull request by konstin**
Escape an apostrophe in the venv path name.

Fixes #8947

### CODE REVIEW

File: crates/uv-virtualenv/src/virtualenv.rs [hunk 1]
```diff
@@ -287,23 +287,34 @@ pub(crate) fn create(
...
+                    .simplified()
+                    .to_str()
+                    // SAFETY: `unwrap` is guaranteed to succeed because `location` is an `Utf8PathBuf`.
+                    .unwrap()
+                    // We want our `activate` script to support any posix shell. There's two kind of quotes:
+                    // Single and double quotes. In bash, single quotes must not contain another single
+                    // quote, you can't even escape it (https://linux.die.net/man/1/bash under "QUOTING").
+                    // Double quotes have escaping rules different from shell to shell, which we can't do.
+                    // Bash has `$'\''`, but that's not universal enough.
+                    // We can use implicit string concatenations, be putting the single quote into double
```
Comment (pantheraleo-7) [inline crates/uv-virtualenv/src/virtualenv.rs:310]:
```suggestion
                    // We can use implicit string concatenations, by putting the single quote into double
```

**Comment by samypr100** [comment]
CC @paveldikov for visiblity on the relocatable side

File: crates/uv-virtualenv/src/virtualenv.rs [hunk 2]
```diff
@@ -287,23 +287,34 @@ pub(crate) fn create(
...
+                    // SAFETY: `unwrap` is guaranteed to succeed because `location` is an `Utf8PathBuf`.
+                    .unwrap()
+                    // We want our `activate` script to support any posix shell. There's two kind of quotes:
+                    // Single and double quotes. In bash, single quotes must not contain another single
+                    // quote, you can't even escape it (https://linux.die.net/man/1/bash under "QUOTING").
+                    // Double quotes have escaping rules different from shell to shell, which we can't do.
+                    // Bash has `$'\''`, but that's not universal enough.
+                    // We can use implicit string concatenations, by putting the single quote into double
+                    // quotes:
+                    .replace('\'', r#"'"'"'"#)
```
Comment (paveldikov) [inline crates/uv-virtualenv/src/virtualenv.rs:312]: I think this is implemented somewhere in the codebase already... IIRC it was cited as a `shlex.quote()` analogue?

Comment (samypr100) [reply crates/uv-virtualenv/src/virtualenv.rs:312]: Yea, it's in another crate

https://github.com/astral-sh/uv/blob/489d42a0125b0770bd6ee14f3068191beff0a8dd/crates/uv/src/commands/venv.rs#L420-L432

Comment (konstin) [reply crates/uv-virtualenv/src/virtualenv.rs]: Thank you! Split out to #9055

File: Cargo.toml [hunk 3]
```diff
@@ -159,7 +159,7 @@ tempfile = { version = "3.12.0" }
 textwrap = { version = "0.16.1" }
 thiserror = { version = "1.0.63" }
 tl = { git = "https://github.com/charliermarsh/tl.git", rev = "6e25b2ee2513d75385101a8ff9f591ef51f314ec" }
-tokio = { version = "1.40.0", features = ["fs", "io-util", "macros", "process", "signal", "sync"] }
+tokio = { version = "1.40.0", features = ["fs", "io-util", "macros", "process", "rt", "signal", "sync"] }
```
Comment (charliermarsh) [inline Cargo.toml:162]: Is this intentional?

File: crates/uv-install-wheel/Cargo.toml [hunk 4]
```diff
@@ -30,6 +30,7 @@ uv-platform-tags = { workspace = true }
 uv-pypi-types = { workspace = true }
 uv-trampoline-builder = { workspace = true }
 uv-warnings = { workspace = true }
+uv-shell = { workspace = true }
```
Comment (charliermarsh) [inline crates/uv-install-wheel/Cargo.toml:33]: Nit: Sort please

**Review by charliermarsh — APPROVED** [review]

**Pull request merged**